pipeline/test_etl.py

```diff
@@ -0,0 +1,232 @@
+"""test_etl.py: Unit tests for the ETL process.
+
+This script tests the entire ETL process:
+- Extraction of sales data from the API.
+- Transformation of extracted data.
+- Loading of transformed data into a database.
+"""
+
+import pytest
+from unittest.mock import patch, MagicMock
+import requests
+import pandas as pd
+from extract import get_sales_information
+from transform import (
+    convert_from_unix_to_datetime,
+    convert_date_format,
+    convert_to_full_url,
+    validate_album_and_track,
+    get_genres_from_url,
+    get_release_date_from_url,
+    create_sales_dataframe,
+)
+from load import (
+    get_connection,
+    get_cursor,
+    insert_country,
+    insert_artist,
+    insert_genres,
+)
+
+class TestExtract:
+    """Tests for the extract phase of the ETL process."""
+
+    @patch("extract.requests.get")
+    def test_extract_get_sales_information_success(self, mock_get):
+        """Verify successful data extraction from the API."""
+        mock_response = MagicMock(status_code=200, json=lambda: {"sales": "data"})
+        mock_get.return_value = mock_response
+
+        result = get_sales_information()
+        assert result == {"sales": "data"}
+        mock_get.assert_called_once_with(
+            "https://bandcamp.com/api/salesfeed/1/get_initial", timeout=1000
+        )
+
+    @patch("extract.requests.get")
+    def test_extract_get_sales_information_failure(self, mock_get):
+        """Verify API failure handling."""
+        mock_response = MagicMock(status_code=404)
+        mock_get.return_value = mock_response
+
+        result = get_sales_information()
+        assert result == {}
+        mock_get.assert_called_once()
+
```

Comment:
Maybe we could add a few more tests for extract to handle edge cases e.g. test_extract_get_sales_information_timeout() , test_extract_get_sales_information_connection_error()